copy/copy.go

```diff
@@ -609,6 +609,18 @@ func (ic *imageCopier) copyBlobFromStream(srcStream io.Reader, srcInfo types.Blo
 		inputInfo.Size = -1
 	}
 
+	// === Decompress the layer if it is compressed and decompression is desired
+	// This is currently only used by docker-archive
+	if isCompressed && ic.dest.ShouldDecompressLayers() {
+		logrus.Debugf("Blob will be decompressed")
+		destStream, err = decompressor(destStream)
+		if err != nil {
```

Comment:
There is already a decompression code around `getOriginalLayerCopyWriter`. If at all possible, we should only decompress the data once and share the output between the two consumers, if they exist at the same time.

Reply:
@mtrmac `getOriginalLayerCopyWriter` is only used when the format of the source image is Docker SchemaV1. So, we could have two decompression if the source format image is Docker SchemaV1 and the destination is `docker-archive`.
But I think it's not trivial to change this behavior since getOriginalLayerCopyWriter is already a little bit tricky...
Do you know if SchemaV1 is still widely used?

Reply:
schema1 is still around (and old images will keep using it), but hopefully its will diminish over time… just like using the `repositories` archive format :)

I really don’t like decompressing the possibly gigabytes of data twice, but you’re right that the code is tricky and I don’t quite have the extra time necessary to get this right, so I guess let’s live with the two separate decompressions.